soroban-env-host/src/host/conversion.rs

```diff
@@ -294,4 +294,27 @@ impl Host {
             )),
         }
     }
+
+    /// Converts a binary search result into a u64. `res` is `Some(index)`
+    /// if the value was found at `index`, or `Err(index)` if the value was not found
+    /// and would've needed to be inserted at `index`.
+    /// Returns a Some(res_u64) where :
+    /// - the high-32 bits is 0x0001 if element existed or 0x0000 if it didn't
+    /// - the low-32 bits contains the u32 representation of the `index`
+    /// Err(_) if the `index` fails to be converted to an u32.
```

Comment:
I hope the documentation of the u64 return value here is good enough. It does feel a bit ad-hoc compared to the standard `Val` encoding we currently have.

Reply:
I agree, but there's nothing here we can't change. @graydon will probably have some good ideas for how we can revisit it.